libs/BLEKit/include/CoreGap.h

```diff
@@ -13,6 +13,15 @@
 
 namespace leka {
 
+struct CustomAdvertising {
+	const char *name;
+	uint8_t battery;
+	uint8_t is_charging;
+	uint8_t led_red;
+	uint8_t led_green;
+	uint8_t led_blue;
+};
```

Comment:
to follow on the previous comment about SRP, you could do something like this:

```suggestion
struct AdvertisingData {
	const char *name;
	uint8_t battery;
	uint8_t is_charging;
	RGB color;
    
	auto getData() -> std::span <uint8_t> {
		auto tmp  = std::to_array<uint8_t> ({battery, is_charging, color.red, color.green, color.blue});
		std::copy(tmp.begin(), tmp.end(), data.begin());
		return std::span<uint8_t> {data.data(), std::size(tmp)};
	}
		
	private:
		std::array<uint8_t, 16> data {};
};
```

see https://gcc.godbolt.org/z/GK7P6KhG6

Reply:
this encapsulate the advertising data logic inside the struct making it easy to change with minimum (if any) impact on the code using it.

Reply:
`getData` will replace a part of process in `setAdvertisingData` in `CoreGap` and its indeed a good way to set data without knowing them in CoreGap

Since there is a function in the struct, I will maybe move it in another file + add unit tests to check its function

Also note that due addition of `std::array data`, it is no more possible to initialize a struct using `AdvertisingData{.name = "Leka"};`

Reply:
this has been discussed directly on slack, are we good on the issue?